build/bazel/remote/execution/v2/remote_execution.proto

```diff
@@ -1839,6 +2010,45 @@ message CacheCapabilities {
   // [BatchUpdateBlobs][build.bazel.remote.execution.v2.ContentAddressableStorage.BatchUpdateBlobs]
   // requests.
   repeated Compressor.Value supported_batch_update_compressors = 7;
+
+  // The maximum size of a blob that may be uploaded into the Content
+  // Addressable Storage if the BLAKE3CONCAT digest function is used.
+  //
+  // Larger blobs may only be formed by first uploading as many blobs as
+  // possible having the maximum size, optionally uploading a blob
+  // containing trailing data, and calling
+  // [ConcatenateBlobs][build.bazel.remote.execution.v2.ContentAddressableStorage.ConcatenateBlobs].
+  //
+  // If supported, this field MUST have value 2^k, where k > 10. It may
+  // not exceed `max_batch_total_size_bytes`.
+  int32 blake3concat_max_upload_size_bytes = 8;
+
+  // The minimum size of blobs that can be created by calling
+  // [SplitBlobs][build.bazel.remote.execution.v2.ContentAddressableStorage.SplitBlobs]
+  // against a blob that uses the BLAKE3CONCAT digest function,
+  // disregarding the blob containing trailing data.
+  //
+  // If supported, this field MUST have value 2^k, where k > 10. It may
+  // not exceed `max_batch_total_size_bytes`.
+  int32 blake3concat_min_split_size_bytes = 9;
```

Comment:
Will the server be able to have an efficient implementation if the API is called with different chunk sizes? Would there be an "optimal" value for a static chunk size to reduce the risk of inefficient client-server combinations?

Reply:
I think that in the case of Buildbarn I will just set `blake3concat_max_upload_size_bytes == blake3concat_min_split_size_bytes`. Then all clients/servers will be in full agreement on what the chunk size is.

The reason I kept them apart, is because I tried to keep the upload & download paths separated. I can imagine that if someone operating a cluster discovers that they made a bad choice regarding chunk size, that having these separate will make it easier to migrate from one chunk size to the other if needed.

With regards to a one size fits all optimal chunk size, I'm not sure whether such a value exists. I guess it really depends on bandwidth vs. latency. If you're bandwidth constrained, then smaller chunks is better. Conversely, if latency is high, it may be desirable to set the chunk size higher, so that you need to call into ConcatenateBlobs() and SplitBlobs() less frequently.